### PR TITLE
Indent: Replace b:verilog_indent_modules with more generic g:verilog_…

### PR DESCRIPTION
**Pull request by lewis6991**
…disable_indent.

Added ability to disable indent in interfaces.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -169,11 +169,21 @@ $ git clone https://github.com/vhda/verilog_systemverilog.vim
 ### Indent options
 
 * __`b:verilog_indent_width`__ - Use this variable to override the option `&shiftwidth`.
-* __`b:verilog_indent_modules`__ - Indent code after module declaration.
 * __`b:verilog_indent_preproc`__ - Indent code after preprocessor statements.
 * __`b:verilog_dont_deindent_eos`__ - Keep last `)` of module port declaration indented.
 * __`b:verilog_indent_assign_fix`__ - Indent assignments by fixed amount.
```
Comment (vhda) [inline README.md]: Please remove this empty line. Something is broken in the HTML output in GitHub and I don't know why.

**Comment by vhda** [comment]
I think we should keep backward compatibility with `b:verilog_indent_modules` for now.

The category "class" is essential and "package" would also make sense.
Basically, the idea is: if it is a container that is usually declared alone in a file named after it, then it should be possible to disable indenting. Since it is normal to have a single top-level container per file, then it doesn't make sense to indent it.

Which leaves the question: even with this option enabled for "module", as an example, shouldn't we still indent a module declared inside another module? We don't need an immediate answer to this question, but it still is worth thinking about.

Also, does it make sense to support "all" in this variable?

**Comment by lewis6991** [comment]
We can keep backward compatibility with `b:verilog_indent_modules` for now, but can we deprecate it and set a roadmap to when it will removed?

I will add support for `class`, `package`, `program`, `covergroup` and maybe preprocessor indentation too. This means `b:verilog_indent_preproc` would also be deprecated.

> Which leaves the question: even with this option enabled for "module", as an example, shouldn't we still indent a module declared inside another module?

I think no because it breaks consistency.

I also don't think `all` would make sense here, as by default all indentation will be enabled. If the option was named `g:verilog_enable_indent`, I would say otherwise.

**Comment by vhda** [comment]
I have a milestone for the end of next month to increase the plugin version to 3.0. I'll create an issue to review the variables to be dropped and associate it with that milestone.

> I think no because it breaks consistency.
> 
> I also don't think all would make sense here, as by default all indentation will be enabled. If the option was named g:verilog_enable_indent, I would say otherwise.

Ok, I agree with your point of view in this regard.

**Comment by vhda** [comment]
Variable updates tracking issue created: #61.

@lewis6991 feel free to edit it directly to add more items.

**Comment by lewis6991** [comment]
@vhda Any chance you can review this?

**Comment by vhda** [comment]
@lewis6991 I apologize, but I missed the new commit and was still waiting for your update.

Everything looks good, so I'll just merge it.

**Pull request merged**